### PR TITLE
Include timestamp tlv in Child ID Response only when it is necessary

### PR DESCRIPTION
**Pull request by chshu**
This does not help to pass any certification test, just to make this part specification compliant.

### CODE REVIEW

**Comment by chshu** [comment]
This PR is ready for merge, please help to review.
(I forgot to increase `mRequestTlvs` array size in the previous commit which caused travis check failures)

File: src/core/thread/mle.cpp [hunk 1]
```diff
@@ -2139,10 +2139,6 @@ ThreadError Mle::HandleChildIdResponse(const Message &aMessage, const Ip6::Messa
             mNetif.GetPendingDataset().Set(pendingTimestamp, aMessage, offset + sizeof(tlv), tlv.GetLength());
         }
     }
-    else
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: Above, there is logic to check if the Active Timestamp is `0` or not and either clear or set the Active Dataset appropriately.  If the Active Timestamp and Active Operational Dataset TLVs are included, should we always set the Active Dataset as well?

Comment (chshu) [inline src/core/thread/mle.cpp]: If the Active Timestamp is 0, it means that the parent doesn't have valid Active Operational Dataset, so I add the check logic. Now, including a 0 value Active Timestamp TLV has the same meaning with the case that not including Active Timestamp TLV in previous implementation.

Comment (jwhui) [inline src/core/thread/mle.cpp]: Yes, I understand.  However, is there any practical value to that?  Whatever Active Dataset the network has (even with a timestamp of 0), the device should synchronize with that to ensure that it has all the same active parameters.

Comment (chshu) [inline src/core/thread/mle.cpp]: Ok, got it. So we will not take 0 as a special timestamp value, I will update the PR, thanks.

**Comment by xiaom-GitHub** [comment]
LGTM 👍

**Pull request merged**